src/nvim/change.c

```diff
@@ -1677,9 +1677,14 @@ int open_line(
         truncate_spaces(saved_line);
       }
       ml_replace(curwin->w_cursor.lnum, saved_line, false);
-      extmark_splice_cols(
-          curbuf, (int)curwin->w_cursor.lnum,
-          0, curwin->w_cursor.col, (int)STRLEN(saved_line), kExtmarkUndo);
+
+      int new_len = (int)STRLEN(saved_line);
+
+      if (new_len != curwin->w_cursor.col) {
```

Comment:
Hmm is it obvious when `curwin->w_cursor.col == new_len` then the bytes is the same? And when they are different, we don't know if there is a shared prefix/suffix?

Reply:
Good question. The new content should be the old content with striped trailing whitespace, thus we have a shared prefix.

Reply:
is this true even with exandtab and such? sometimes vim will change tabs to spaces and back automagically.

Reply:
BTW, if you don't want to look into this right now, just put a `TODO` here.